### PR TITLE
Support upstream MPTCP

### PR DESCRIPTION
**Pull request by serrislew**
MPTCP is now supported in upstream linux kernels. Allow ATS to use `IPPROTO_MPTCP` if configured
- for ATS logging, 5.16+ kernel needed to detect mptcp_info struct

### CODE REVIEW

**Review by bneradt — CHANGES_REQUESTED** [review]
I assume we'll need a change to CMakeLists.txt for cmake builds now that that's the supported build system for master. Or am I missing something? Be aware that all of PR CI now runs cmake for master, so it will not cover anything done in configure.ac.

File: mgmt/rpc/overridable_txn_vars.cc [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "../../src/shared/overridable_txn_vars.cc"
```
Comment (cmcfarlen) [inline mgmt/rpc/overridable_txn_vars.cc:1]: The mgmt code moved to src/mgmt, but this file shouldn't be necessary regardless.  Please remove.

Comment (serrislew) [reply mgmt/rpc/overridable_txn_vars.cc:1]: Oh I don't even remember including this.. will remove

**Comment by bneradt** [comment]
[approve ci clang-format]

**Review by bneradt — APPROVED** [review]

**Comment by bneradt** [comment]
[approve ci clang-format]

**Pull request merged**